File: N/2437-number-of-valid-clock-times/main.go
package main

func countTime(time string) int {
	hour := 1
	minutes := 1
	if time[0] == '?' && time[1] == '?' {
		hour = 24
	} else if time[0] == '?' {
		if time[1] <= '3' && time[1] >= '0' {
			hour = 3
		} else {
			hour = 2
		}
	} else if time[1] == '?' {
		if time[0] == '2' {
			hour = 4
		} else {
			hour = 10
		}
	}

	if time[3] == '?' {
		minutes *= 6
	}
	if time[4] == '?' {
		minutes *= 10
	}
	return hour * minutes
}

func main() {

}

File: N/2437-number-of-valid-clock-times/main.rs
struct Solution {}

impl Solution {
  pub fn count_time(time: String) -> i32 {
    let mut hours: i32 = 1;
    let mut minutes: i32 = 1;
    let question: u8 = '?' as u8;
    let bb: Vec<u8> = time.as_bytes().to_vec();
    if bb[0] == question && bb[1] == question {
      hours = 24;
    } else if bb[0] == question {
      if bb[1] <= '3' as u8 && bb[1] >= '0' as u8 {
        hours = 3
      } else {
        hours = 2
      }
    } else if bb[1] == question {
      if bb[0] == '2' as u8 {
        hours = 4
      } else {
        hours = 10
      }
    }

    if bb[3] == question {
      minutes *= 6
    }
    if bb[4] == question {
      minutes *= 10
    }
    hours * minutes
  }
}
